### PR TITLE
add new option remove_duplicates to ListField and RelatedListField

### LINKED ISSUE #1017: Add remove_duplicates option to ListField/RelatedListField
## Issue description

We give two main options for storing arrays of data on a model, these are `ListField` and `SetField`. The advantage of `ListField` is that it remains ordered, whereas the advantage of `SetField` is that it's not possible that it can contain duplicate entries.

However, sometimes you need order, but also to not duplicate values (in fact, that's fairly common in `ListFields`). It's simple enough to remove duplicates and retain order so I suggest we add a `remove_duplicates` keyword argument to `ListField` (defaulting to `False`) which when saving will remove any duplicated elements. 

We should also add this to `RelatedListField` if we are adding it to `ListField`

### PR DESCRIPTION
**Pull request by michal-macioszczyk**
Fixes #1017

Summary of changes proposed in this Pull Request:
- add new kwargs option "remove_duplicates" to ListField and RelatedListField
- fix some PEP-8 warnings 

PR checklist:
- [x] Updated relevant documentation
- [x] Updated CHANGELOG.md 
- [x] Added tests for my change

### CODE REVIEW

File: djangae/db/utils.py [hunk 1]
```diff
@@ -473,3 +473,12 @@ def ensure_datetime(value):
     if isinstance(value, long):
         return datetime.fromtimestamp(value / 1e6)
     return value
+
+
+def remove_duplicates_form_list(list_to_scan):
+    remove_duplicates = []
+    for idx in xrange(len(list_to_scan)):
```
Comment (davidwtbuxton) [inline djangae/db/utils.py]: xrange has been removed in Python 3.

Why not use `for el in list_to_scan`? We don't need the index.

File: djangae/db/utils.py [hunk 2]
```diff
@@ -473,3 +473,12 @@ def ensure_datetime(value):
     if isinstance(value, long):
         return datetime.fromtimestamp(value / 1e6)
     return value
+
+
+def remove_duplicates_form_list(list_to_scan):
+    remove_duplicates = []
```
Comment (davidwtbuxton) [inline djangae/db/utils.py]: Would "unique_items" be a better name for this list?

File: djangae/db/utils.py (hunk 1 above)
Comment (michal-macioszczyk) [reply djangae/db/utils.py]: ah good spot ! 👍

File: djangae/db/utils.py (hunk 2 above)
Comment (michal-macioszczyk) [reply djangae/db/utils.py]: yes it is better name 👍

File: djangae/db/utils.py [hunk 3]
```diff
@@ -473,3 +473,12 @@ def ensure_datetime(value):
     if isinstance(value, long):
         return datetime.fromtimestamp(value / 1e6)
     return value
+
+
+def remove_duplicates_form_list(list_to_scan):
```
[the remark this answers is not in the document]
Comment (michal-macioszczyk) [reply djangae/db/utils.py]: 👍

**Pull request merged**